mmf/trainers/core/evaluation_loop.py

```diff
@@ -17,33 +17,53 @@
 
 class TrainerEvaluationLoopMixin(ABC):
     def evaluation_loop(
-        self, loader, use_tqdm: bool = False, single_batch: bool = False
+        self, dataset_type: str, use_tqdm: bool = False, single_batch: bool = False
     ) -> Tuple[Dict[str, Any], Type[Meter]]:
         meter = Meter()
+        reporter = self.dataset_loader.get_test_reporter(dataset_type)
 
         with torch.no_grad():
             self.model.eval()
             disable_tqdm = not use_tqdm or not is_master()
-            combined_report = None
 
-            for batch in tqdm.tqdm(loader, disable=disable_tqdm):
-                report = self._forward(batch)
-                self.update_meter(report, meter)
+            while reporter.next_dataset(flush_report=False):
+                dataloader = reporter.get_dataloader()
+
+                combined_report = None
+                for batch in tqdm.tqdm(dataloader, disable=disable_tqdm):
+                    prepared_batch = reporter.prepare_batch(batch)
+                    model_output = self.model(prepared_batch)
+                    report = Report(prepared_batch, model_output)
```

Comment:
Once the above mentioned change happens, please move back to using `self._forward`.